mlir-tensorrt/compiler/include/mlir-tensorrt-c/Compiler/Compiler.h

```diff
@@ -51,6 +51,28 @@ static inline bool mtrtCompilerClientIsNull(MTRT_CompilerClient options) {
   return !options.ptr;
 }
 
+//===----------------------------------------------------------------------===//
+// MTRT_OptionsContext
+//===----------------------------------------------------------------------===//
+
+typedef struct MTRT_OptionsContext {
+  void *ptr;
+} MTRT_OptionsContext;
+
+MLIR_CAPI_EXPORTED MTRT_Status mtrtOptionsContextCreateFromArgs(
+    MTRT_CompilerClient client, MTRT_OptionsContext *options,
+    const char *optionsType, const MlirStringRef *argv, unsigned argc);
+
+MLIR_CAPI_EXPORTED MTRT_Status
+mtrtOptionsContextPrint(MTRT_OptionsContext options);
```

Comment:
There's a simple uniform convention we use for print methods. Just look for the other C API methods that accept a `MlirStringCallback`.

Here's how to do it:

```
MLIR_CAPI_EXPORTED void mlirOptionsContextPrint(MTRT_OptionsContext options, MlirStringCallback append, void *userCtx);
```

In CAPI implementation:

```C++
void mlirOptionsContextPrint(MTRT_OptionsContext options, MlirStringCallback append, void *userCtx) { 
    mlir::detail::CallbackOstream stream(callback, userData);
    unwrap(options)->print(stream);
}
```

In PyBind:

```C++
.def("__repr__", [](PyOptionsContext &ctx)  {
               PyPrintAccumulator printAccum;
             printAccum.parts.append("Options(");
             mtrtOptionsContextPrint(self, printAccum.getCallback(),
                                 printAccum.getUserData());
             printAccum.parts.append(")");
             return printAccum.join();
}
```